src/pytorch_lightning/strategies/ipu.py

```diff
@@ -272,8 +274,13 @@ def to_tensor(x):
     def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
         # This override is necessary because the cast must occur before the data
         # is moved to the device to prevent wasteful host->device copies.
+        def fp_to_half(tensor: Tensor) -> Tensor:
+            if torch.is_floating_point(tensor):
+                return tensor.half()
+            return tensor
+
         if self.precision_plugin.precision in (PrecisionType.MIXED, PrecisionType.HALF):
-            batch = apply_to_collection(batch, Tensor, function=Tensor.half)
+            batch = apply_to_collection(batch, Tensor, function=fp_to_half)
```

Comment:
```suggestion
            batch = apply_to_collection(batch, Tensor, function=self.batch_to)
```

Reply:
that's a different deprecated class that is not used anymore

Reply:
true! sorry, didn't unfold the wrapped code.